### PR TITLE
CategoricalCrossEntropy: support missing values for int arrays

### PR DESCRIPTION
**Pull request by danieldk**
`CategoricalCrossEntropy` provides a `missing_value` option that can be used
to mask instances that have a special label. However, this functionality
only works with non-integer categorical labels.

This change adds support for masking missing values for int lists and xp
arrays.

### CODE REVIEW

File: thinc/loss.py [hunk 1]
```diff
@@ -342,6 +351,22 @@ def _make_mask(missing, guesses) -> Floats2d:
     return mask
 
 
+def _make_xp_mask(guesses, truths, missing_value) -> Floats2d:
```
Comment (svlandeg) [inline thinc/loss.py:354]: This looks good to me in general, but it annoys me slightly that the signature of `_make_xp_mask` is so different to `_make_mask` in the order of the arguments. Maybe `_make_mask` should have had the order of arguments reversed, and it is a private function so maybe it wouldn't even hurt to change this on the fly, just to have some kind of consistency between the function declarations?

Comment (svlandeg) [reply thinc/loss.py:354]: We should probably also reconsider the naming of these two functions...

Comment (danieldk) [reply thinc/loss.py]: I changed the order of `_make_xp_mask` to correspond to `convert_truths`.

Do you have suggestions for the name? Maybe `_missing_value_mask`?

Comment (svlandeg) [reply thinc/loss.py]: `_make_mask_by_value`?

Comment (danieldk) [reply thinc/loss.py]: 👍🏻 Done!

Comment (svlandeg) [reply thinc/loss.py]: Oh hm wait, I'm confused, I thought we'd have `_make_mask` as before and `_make_mask_by_value` as the latter checks on the missing value?

Comment (danieldk) [reply thinc/loss.py]: Makes sense! I'll update the names.

Comment (danieldk) [reply thinc/loss.py]: I did do a force push this time, to avoid a commit which fixes a commit ;).

**Pull request merged**